### PR TITLE
feat: intercom group support

### PR DESCRIPTION
**Pull request by aashishmalik**
## Description of the change

> intercom group support

## Type of change
- [ ] Bug fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Related issues

> Fix [#1]() 

### Development

- [ ] Lint rules pass locally
- [ ] The code changed/added as part of this pull request has been covered with tests
- [ ] All tests related to the changed code pass in development

### Code review 

- [ ]  This pull request has a descriptive title and information useful to a reviewer. There may be a screenshot or screencast attached
- [ ] "Ready for review" label attached to the PR and reviewers mentioned in a comment
- [ ] Changes have been reviewed by at least one other engineer
- [ ] Issue from task tracker has a link to this pull request

### CODE REVIEW

File: src/v0/destinations/intercom/data/INTERCOMGroupConfig.json [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+  },
+  {
+    "destKey": "custom_attributes",
+    "sourceKeys": ["traits"],
+    "required": false,
+    "metadata": {
+      "excludes": ["groupId"]
+    }
+  }
+]
```
Comment (ItsSudip) [inline src/v0/destinations/intercom/data/INTERCOMGroupConfig.json:59]: We will not get name, plan, size etc in the root of the payload. We will get them inside traits.

File: src/v0/destinations/intercom/transform.js [hunk 2]
```diff
@@ -89,30 +90,72 @@ function validateTrack(message, payload) {
...
+  const requestBody = {};
+  if (message.userId) {
+    requestBody.user_id = message.userId;
+  }
+  if (message.email) {
+    requestBody.email = message.email;
+  }
+  const companyObj = {};
+  companyObj.company_id = message.groupId;
+  companyObj.name = message.name;
```
Comment (ItsSudip) [inline src/v0/destinations/intercom/transform.js:106]:
```suggestion
  companyObj.name = message.traits.name;
```

File: src/v0/destinations/intercom/transform.js [hunk 3]
```diff
@@ -89,30 +90,72 @@ function validateTrack(message, payload) {
...
+      response.body.JSON = removeUndefinedAndNullValues(payload);
+      respList.push(response);
+      try {
+        response = attachUserAndCompany(message, destination.Config);
+        if (response) {
+          response.userId = message.anonymousId;
+          respList.push(response);
+        }
+      } catch (exp) {
+        logger.info("failed to send /users call in group event", exp);
```
Comment (ItsSudip) [inline src/v0/destinations/intercom/transform.js:151]: Do we need this?

**Review by ItsSudip — CHANGES_REQUESTED** [review]

File: src/v0/destinations/intercom/transform.js (hunk 3 above)
Comment (aashishmalik) [reply src/v0/destinations/intercom/transform.js:151]: I added this log to to catch any discrepency when linking user and company

File: src/v0/destinations/intercom/transform.js (hunk 2 above)
Comment (aashishmalik) [reply src/v0/destinations/intercom/transform.js]: done

File: src/v0/destinations/intercom/data/INTERCOMGroupConfig.json [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+    "destKey": "remote_created_at",
+    "sourceKeys": ["traits.remoteCreatedAt","context.traits.remoteCreatedAt"],
+    "metadata": {
+      "type": "toNumber"
+    },
+    "required": false
+  },
+  {
+    "destKey": "custom_attributes",
+    "sourceKeys": ["traits.customAttributes","context.traits.customAttributes"],
```
Comment (ItsSudip) [inline src/v0/destinations/intercom/data/INTERCOMGroupConfig.json:53]: We shouldn't expect customAttributes like this. We should put all the traits not on our mapping to custom_attributes by default.

Comment (aashishmalik) [reply src/v0/destinations/intercom/data/INTERCOMGroupConfig.json]: updated

**Review by ItsSudip — APPROVED** [review]

**Comment by anantjain45823** [comment]
Can we use generic mapping in Configs?

**Review by anantjain45823 — APPROVED** [review]

**Review by ItsSudip — APPROVED** [review]

**Review by ItsSudip — APPROVED** [review]

**Review by mihir-4116 — APPROVED** [review]

**Pull request merged**